File: apps/web/src/services/incidentService.ts
import apiClient from './apiClient';
import { Incident, IncidentSeverity, IncidentStatus } from '../types';

export interface CreateIncidentPayload {
  title: string;
  description: string;
  severity: IncidentSeverity;
  category: string;
}

export interface UpdateStatusPayload {
  status: IncidentStatus;
  assignedAnalyst?: string;
}

export const incidentService = {
  async getAll(): Promise<Incident[]> {
    const { data } = await apiClient.get<Incident[]>('/incidents');
    return data;
  },

  async getById(id: string): Promise<Incident> {
    const { data } = await apiClient.get<Incident>(`/incidents/${id}`);
    return data;
  },

  async create(payload: CreateIncidentPayload): Promise<Incident> {
    const { data } = await apiClient.post<Incident>('/incidents', payload);
    return data;
  },

  async updateStatus(id: string, payload: UpdateStatusPayload): Promise<Incident> {
    const { data } = await apiClient.patch<Incident>(`/incidents/${id}/status`, payload);
    return data;
  },
};

File: apps/api/Services/IncidentService.cs
using SCIP.Api.DTOs;
using SCIP.Api.Entities;

namespace SCIP.Api.Services
{
    public class IncidentService : IIncidentService
    {
        private readonly ScipDbContext _dbContext;

        public IncidentService(ScipDbContext dbContext)
        {
            _dbContext = dbContext;
        }

        public IEnumerable<IncidentResponseDto> GetAllIncidents()
        {
            return _dbContext.Incidents
                .Select(i => new IncidentResponseDto(
                    i.Id, i.Title, i.Description, i.Severity, i.Status,
                    i.Category, i.ReportedBy, i.EvidenceCount,
                    i.CreatedAt, i.UpdatedAt
                ))
                .ToList();
        }

        public IncidentResponseDto? GetIncidentById(Guid id)
        {
            return _dbContext.Incidents
                .Where(i => i.Id == id)
                .Select(i => new IncidentResponseDto(
                    i.Id, i.Title, i.Description, i.Severity, i.Status,
                    i.Category, i.ReportedBy, i.EvidenceCount,
                    i.CreatedAt, i.UpdatedAt
                ))
                .FirstOrDefault();
        }

        public IncidentResponseDto CreateIncident(CreateIncidentDto dto, string reportedBy)
        {
            var incident = new Incident
            {
                Id = Guid.NewGuid(),
                Title = dto.Title,
                Description = dto.Description,
                Severity = dto.Severity,
                Status = IncidentStatus.New,
                Category = dto.Category,
                ReportedBy = reportedBy,
                EvidenceCount = 0,
                CreatedAt = DateTime.UtcNow,
                UpdatedAt = DateTime.UtcNow
            };

            _dbContext.Incidents.Add(incident);
            _dbContext.SaveChanges();

            return new IncidentResponseDto(
                incident.Id, incident.Title, incident.Description,
                incident.Severity, incident.Status, incident.Category,
                incident.ReportedBy, incident.EvidenceCount,
                incident.CreatedAt, incident.UpdatedAt
            );
        }

        public IncidentResponseDto? UpdateStatus(Guid id, UpdateIncidentStatusDto dto)
        {
            var incident = _dbContext.Incidents.Find(id);
            if (incident == null) return null;

            incident.Status = dto.Status;
            incident.AssignedAnalyst = dto.AssignedAnalyst ?? incident.AssignedAnalyst;
            incident.UpdatedAt = DateTime.UtcNow;

            _dbContext.SaveChanges();

            return new IncidentResponseDto(
                incident.Id, incident.Title, incident.Description,
                incident.Severity, incident.Status, incident.Category,
                incident.ReportedBy, incident.EvidenceCount,
                incident.CreatedAt, incident.UpdatedAt
            );
        }
    }
}
